admin/installer/cloudformation.py

```diff
@@ -0,0 +1,293 @@
+desired stack's resources and properties:
+1 Control Node with Flocker Control Service, (TLS-enabled) Swarm Manager,
+                    (TLS-enabled) Docker, Ubuntu 14.04
+2 Agent Nodes with Flocker Dataset Agent, Swarm Agent, (TLS-enabled) Docker,
+                   Ubuntu 14.04
+1 Client Node with Flockerctl, Docker, Docker-compose, Ubuntu 14.04
+
+To manifest the blueprint, please input the JSON template at AWS CloudFormation
+Create Stack console (after replacing ``us-east-1`` with your Region):
+https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/new
+
+"""
+
+import json
+import os
+import re
+import urllib
+
+from troposphere import FindInMap, GetAtt, Base64, Join
+from troposphere import Parameter, Output, Ref, Template, GetAZs, Select
+from troposphere.s3 import Bucket
+import troposphere.ec2 as ec2
+from troposphere.cloudformation import WaitConditionHandle, WaitCondition
+
+NUM_NODES = 3
+AGENT_NODE_NAME_TEMPLATE = u"AgentNode{index}"
+CONTROL_NODE_NAME = u"ControlNode"
+CLIENT_NODE_NAME = u"ClientNode"
+INFRA_WAIT_HANDLE_TEMPLATE = u"{node}FlockerSwarmReadySignal"
+INFRA_WAIT_CONDITION_TEMPLATE = u"{node}FlockerSwarmSetup"
+CLIENT_WAIT_HANDLE = u"ClientReadySignal"
+CLIENT_WAIT_CONDITION = u"ClientSetup"
+S3_SETUP = 'setup_s3.sh'
+DOCKER_SETUP = 'setup_docker.sh'
+DOCKER_SWARM_CA_SETUP = 'docker-swarm-ca-setup.sh'
+SWARM_MANAGER_SETUP = 'setup_swarm_manager.sh'
+SWARM_NODE_SETUP = 'setup_swarm_node.sh'
+FLOCKER_CONFIGURATION_GENERATOR = 'flocker-configuration-generator.sh'
+FLOCKER_CONFIGURATION_GETTER = 'flocker-configuration-getter.sh'
+CLIENT_SETUP = 'setup_client.sh'
+SIGNAL_CONFIG_COMPLETION = 'signal_config_completion.sh'
+
+
+def _sibling_lines(filename):
+    """
+    Read file content into an output string.
+    """
+    dirname = os.path.dirname(__file__)
+    path = os.path.join(dirname, filename)
+    with open(path, 'r') as f:
+        return f.readlines()
+
+# Base JSON template.
+template = Template()
+
+# Keys corresponding to CloudFormation user Inputs.
+keyname_param = template.add_parameter(Parameter(
+    "KeyName",
+    Description="Name of an existing EC2 KeyPair to enable SSH "
+                "access to the instance",
+    Type="String",
+))
+access_key_id_param = template.add_parameter(Parameter(
+    "AccessKeyID",
+    Description="Your Amazon AWS access key ID",
+    Type="String",
+))
+secret_access_key_param = template.add_parameter(Parameter(
+    "SecretAccessKey",
+    Description="Your Amazon AWS secret access key.",
+    Type="String",
+))
+
+# Base AMIs.
+template.add_mapping(
+    'RegionMap', {
+        'us-east-1':      {"FlockerAMI": "ami-5f401635",
+                           "ClientAMI": "ami-86590fec"},
+        'us-west-1':      {"FlockerAMI": "ami-e83a5088",
+                           "ClientAMI": "ami-0a254f6a"},
+        'us-west-2':      {"FlockerAMI": "ami-bd2b35dc",
+                           "ClientAMI": "ami-fa2b359b"},
```

Comment:
@wallrj is working on adding Packer AMIs for all non-US regions in his Packer branch (https://github.com/ClusterHQ/flocker/pull/2451). Added a comment about the work in progress. Thanks!